Dev/neo4j

kevsahn opened the pull request:
1. Neo4j 工具类 (neo4j_util.py)：连接管理、CRUD操作、搜索功能
  2. 图服务层 (graph_service.py)：业务逻辑、关系管理、推荐算法
  3. API 端点 (graph.py)：RESTful接口、测试端点、业务端点
  4. Docker 配置：Neo4j容器、端口7474/7687、数据持久化
  

Neo4j用户名：neo4j
Neo4j密码：password123

Commit:
引入neo4j，添加相关配置，创建工具类服务层和api断点，实现简单的搜索修改功能

Commit:
引入neo4j，添加相关配置，创建工具类服务层和api断点，实现简单的搜索修改功能

kevsahn committed:
修复错误的import

kevsahn committed:
修复错误的import

jaywang5540 commented on app/api/v1/endpoints/graph.py:
```diff
@@ -0,0 +1,249 @@
+            }
+        }
+    """
+    try:
```
可以主要依赖fastapi的异常处理，所以不用try以后再返回500错误了

jaywang5540 commented on docker-compose.yml:
```diff
@@ -22,4 +22,23 @@ services:
+    restart: always
+
+  neo4j:
+    image: neo4j:5-community
```
这里咱们选一个release的版本
neo4j:5.15

jaywang5540 committed:
refactor: 优化服务依赖注入，使用 lru_cache 替代单例模式

- 移除 LangGraphService 和 EmbeddingService 的手动单例实现
- 统一使用 @lru_cache() 装饰器进行应用级缓存
- ConversationService 改为通过依赖注入接收 LangGraphService
- 保持有状态组件（MemorySaver）的跨请求持久性

jaywang5540 committed:
refactor: 优化服务依赖注入，使用 lru_cache 替代单例模式

- 移除 LangGraphService 和 EmbeddingService 的手动单例实现
- 统一使用 @lru_cache() 装饰器进行应用级缓存
- ConversationService 改为通过依赖注入接收 LangGraphService
- 保持有状态组件（MemorySaver）的跨请求持久性

jaywang5540 committed:
refactor: 统一 import 路径为绝对导入

jaywang5540 committed:
refactor: 统一 import 路径为绝对导入

jaywang5540 committed:
refactor(neo4j): 优化Neo4j Repository模式并简化代码注释

- 实现完整的Repository模式架构
- 添加GraphRepository统一接口
- 分离节点和关系操作到独立的Repository
- 简化所有文件的注释和文档字符串
- 添加依赖注入支持
- 清理旧版本代码并统一实现
- 删除不必要的文档文件

jaywang5540 committed:
refactor(neo4j): 优化Neo4j Repository模式并简化代码注释

- 实现完整的Repository模式架构
- 添加GraphRepository统一接口
- 分离节点和关系操作到独立的Repository
- 简化所有文件的注释和文档字符串
- 添加依赖注入支持
- 清理旧版本代码并统一实现
- 删除不必要的文档文件

jaywang5540 committed:
refactor: 统一Response定义并删除冗余代码

- 删除重复的common.py，保留功能更完整的common_response.py
- 重命名CommonResponse.py为common_response.py符合Python命名规范
- 统一所有文件的Response导入路径
- 更新ResponseUtil使用PaginatedResponse类替代字典
- 删除已废弃的neo4j_util.py文件

jaywang5540 committed:
refactor: 统一Response定义并删除冗余代码

- 删除重复的common.py，保留功能更完整的common_response.py
- 重命名CommonResponse.py为common_response.py符合Python命名规范
- 统一所有文件的Response导入路径
- 更新ResponseUtil使用PaginatedResponse类替代字典
- 删除已废弃的neo4j_util.py文件

kevsahn committed:
feat: Add hybrid memory system for AI agents

  - Implement IMemory interface with Chroma (vectors) +
  Neo4j (relationships)
  - Add LLM-based importance, type classification, and
  relationship detection
  - Create entity extraction and co-occurrence tracking
  - Support memory consolidation, decay, and statistics
  - Enable hybrid retrieval: vector search → entity
  discovery → graph expansion

kevsahn committed:
feat: Add hybrid memory system for AI agents

  - Implement IMemory interface with Chroma (vectors) +
  Neo4j (relationships)
  - Add LLM-based importance, type classification, and
  relationship detection
  - Create entity extraction and co-occurrence tracking
  - Support memory consolidation, decay, and statistics
  - Enable hybrid retrieval: vector search → entity
  discovery → graph expansion

kevsahn committed:
Merge branch 'dev/neo4j'

# Conflicts:
#	app/api/v1/endpoints/conversations.py
#	app/dependencies/services.py
#	app/services/conversation_service.py

kevsahn committed:
update with main branch

kevsahn committed:
Merge branch 'main' into dev/neo4j

jaywang5540 committed:
Merge remote-tracking branch 'origin/dev/neo4j' into dev/neo4j

# Conflicts:
#	app/api/v1/endpoints/conversations.py

jaywang5540 committed:
refactor(static): 整合所有静态资源到单文件

- 将 chroma-test 的 CSS 和 JS 内嵌到 HTML 中
- 删除独立的 CSS 和 JS 文件
- 删除未使用的 index.html 及相关文件
- 简化静态资源结构，便于维护和分发

jaywang5540 committed:
refactor(static): 整合所有静态资源到单文件

- 将 chroma-test 的 CSS 和 JS 内嵌到 HTML 中
- 删除独立的 CSS 和 JS 文件
- 删除未使用的 index.html 及相关文件
- 简化静态资源结构，便于维护和分发

jaywang5540 committed:
feat(graph): 添加Neo4j图数据库Pydantic模型系统

- 实现Node和Relationship基础类，支持Neo4j数据转换
- 添加PersonNode和OrganizationNode节点模型
- 实现社交和职业关系模型(Friend, Family, WorksAt等)
- 创建GraphModelFactory支持动态模型创建
- 增强Repository层，提供类型安全的数据库操作
- 所有模型兼容Pydantic V2，支持AI工具调用集成

jaywang5540 committed:
feat(graph): 添加Neo4j图数据库Pydantic模型系统

- 实现Node和Relationship基础类，支持Neo4j数据转换
- 添加PersonNode和OrganizationNode节点模型
- 实现社交和职业关系模型(Friend, Family, WorksAt等)
- 创建GraphModelFactory支持动态模型创建
- 增强Repository层，提供类型安全的数据库操作
- 所有模型兼容Pydantic V2，支持AI工具调用集成

kevsahn committed:
Merge branch 'dev/neo4j' of https://github.com/soontechgroup/ai-agents into dev/neo4j

jaywang5540 committed:
fix(graph): 修复Pydantic V2中Field参数错误

将所有max_items参数替换为max_length，以兼容Pydantic V2的API变更

jaywang5540 committed:
fix(graph): 修复Pydantic V2中Field参数错误

将所有max_items参数替换为max_length，以兼容Pydantic V2的API变更

jaywang5540 committed:
refactor(graph): 将所有相对导入改为绝对导入

统一使用app.models.graph前缀的绝对路径导入，提高代码可维护性和模块独立性

jaywang5540 committed:
refactor(graph): 将所有相对导入改为绝对导入

统一使用app.models.graph前缀的绝对路径导入，提高代码可维护性和模块独立性

jaywang5540 committed:
refactor: 重构Neo4j架构到Neomodel ORM并统一API为POST请求

主要变更：
- 从neo4j-driver迁移到Neomodel ORM框架
- 实现Pydantic与Neomodel双向模型转换
- 重新设计API为POST-only路径模式
- 移除未使用的推荐系统功能
- 重组请求模型到schemas目录

新增功能：
- Neomodel配置和初始化系统
- 基于Neomodel的完整仓储层
- 业务导向的API端点（人员、组织、关系、分析）
- 模型转换器实现Pydantic↔Neomodel转换

架构改进：
- 清晰的分层架构：API层→服务层→仓储层→ORM层
- 更好的代码组织和关注点分离
- 统一的POST请求处理模式
- 规范的schemas目录结构

jaywang5540 committed:
feat: 添加Neo4j人物关系网络测试页面及修复neomodel集成问题

- 创建neo4j-person-network.html可视化测试页面
- 修复neomodel保留字冲突(source->data_source)
- 修复NEO4J配置属性解析问题
- 注释未实现的模型导入避免启动错误
- 更新neomodel版本至5.5.0

jaywang5540 committed:
feat(neo4j): 完善图数据库功能和可视化界面

- 移除测试端点，统一使用正式API
- 所有图数据库端点暂时禁用认证
- 修复PersonNode的uid字段为可选
- 添加清空所有数据功能
- 优化测试数据生成为单个添加模式
- 修复前端API调用和数据显示问题
- 移除硬编码示例数据，改为加载真实数据

jaywang5540 committed:
fix(neo4j): 修复人物关系网络页面和API接口问题

- 修复vis-network配置错误(arrows改为对象格式)
- 修复nodes.update()参数格式(改为数组)
- 修复人员网络查询(支持FRIEND_OF关系类型)
- 完善关系创建API的UID/名称兼容处理
- 添加neo4j数据目录到.gitignore

jaywang5540 committed:
fix(neo4j): 修复关系创建后刷新页面关系丢失的问题

- 添加API响应状态检查，确保关系创建成功后才在前端显示
- generateTestData函数中的自动关系创建改为调用后端API持久化
- generateRelationships函数改为真正调用后端API创建关系
- 创建关系后自动重新加载关系数据确保同步

jaywang5540 committed:
refactor(ui): 移除Neo4j页面的分析模块并修复缺失函数

- 移除分析Tab按钮和整个分析内容区域
- 删除所有分析相关的JavaScript函数(analyzePerson、findPath等)
- 添加缺失的图谱事件处理函数(handleNodeClick等)
- 恢复loadGraphData和updateGraphDisplay函数
- 添加图谱控制功能函数
- 删除不再需要的文档和示例文件

kevsahn committed:
Merge remote-tracking branch 'refs/remotes/origin/dev/neo4j' into dev/neo4j

kevsahn committed:
Merge branch 'main' into dev/neo4j

kevsahn committed:
clean up commit

jaywang5540 committed:
Merge remote-tracking branch 'origin/dev/neo4j' into dev/neo4j

kevsahn committed:
feat: 实现基于Neo4j的知识图谱存储系统

  - 使用Neomodel ORM创建Knowledge和Entity节点模型
  - 实现知识分类：事实、经验、偏好、技能、规则、概念
  - 添加知识验证状态和置信度评分机制
  - 创建实体提取和共现关系追踪功能
  - 实现知识仓库层，支持CRUD操作和图查询
  - 添加知识关系管理：MENTIONS、RELATED_TO、CO_OCCURS
  - 编写完整测试脚本验证所有功能
  - 创建详细实现文档，包含架构设计和后续开发计划

jaywang5540 committed:
refactor: 重构图数据库服务层消除静态方法警告

- 创建GraphRepository处理跨节点类型的图查询
- 重构GraphService中的list_relationships、get_system_statistics、find_shortest_path方法
- 修改create_person和create_organization使用仓储层
- 修复relationships.py的架构违规问题
- 删除未使用的批量导入功能(import_persons_batch)
- 清理测试文件
- 修复MySQL连接配置(172.17.0.1改为localhost)

jaywang5540 committed:
refactor(api): 合并记忆体系统API端点

- 创建统一的memory.py管理所有记忆体相关操作
- 将persons、organizations、relationships三个文件合并
- 18个端点整合到一个文件，使用子路由组织
- 更新前端页面API调用路径配合新结构
- 清理GraphService中未使用的导入和TODO注释

架构优化：
- 记忆体系统作为统一领域概念
- 减少文件数量：3个文件→1个文件
- 便于维护和扩展

jaywang5540 committed:
refactor(schemas): 将请求模型从API端点移至schemas目录

- 将memory.py中的所有请求模型定义移至对应的schema文件
- 在relationship.py中新增ListRelationshipsRequest模型
- 更新memory.py的导入语句，从schemas目录导入所有请求模型
- 移除memory.py中的冗余导入和重复定义
